src/winforms/toga_winforms/widgets/webview.py

```diff
@@ -1,13 +1,98 @@
+import sys
+
 from travertino.size import at_least
 
 from toga_winforms.libs import Uri, WinForms
 
 from .base import Widget
 
+ie_updated = False
+
+
+def _set_ie_mode():
+    """
+    By default hosted IE control emulates IE7 regardless which version of IE is installed. To fix this, a proper value
+    must be set for the executable.
+    See http://msdn.microsoft.com/en-us/library/ee330730%28v=vs.85%29.aspx#browser_emulation for details on this
+    behaviour.
+    """
+
+    global ie_updated
+
+    if ie_updated:
+        return
+
+    try:
+        import _winreg as winreg  # Python 2
+    except ImportError:
+        import winreg  # Python 3
+
+    def get_ie_mode():
+        """
+        Get the installed version of IE
+        :return:
+        """
+        ie_key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE,
+                                r"Software\Microsoft\Internet Explorer")
+        try:
+            version, type = winreg.QueryValueEx(ie_key, "svcVersion")
+        except Exception:
+            version, type = winreg.QueryValueEx(ie_key, "Version")
+
+        winreg.CloseKey(ie_key)
+
+        if version.startswith("11"):
+            value = 0x2AF9
+        elif version.startswith("10"):
+            value = 0x2711
+        elif version.startswith("9"):
+            value = 0x270F
+        elif version.startswith("8"):
+            value = 0x22B8
+        else:
+            value = 0x2AF9  # Set IE11 as default
+
+        return value
+
+    try:
+        browser_emulation = winreg.OpenKey(
+            winreg.HKEY_CURRENT_USER,
+            r"Software\Microsoft\Internet Explorer\Main\FeatureControl\FEATURE_BROWSER_EMULATION",
+            0, winreg.KEY_ALL_ACCESS)
+    except WindowsError:
+        browser_emulation = winreg.CreateKeyEx(
+            winreg.HKEY_CURRENT_USER,
+            r"Software\Microsoft\Internet Explorer\Main\FeatureControl\FEATURE_BROWSER_EMULATION",
+            0, winreg.KEY_ALL_ACCESS)
+
+    try:
+        dpi_support = winreg.OpenKey(
+            winreg.HKEY_CURRENT_USER,
+            r"Software\Microsoft\Internet Explorer\Main\FeatureControl\FEATURE_96DPI_PIXEL",
+            0, winreg.KEY_ALL_ACCESS)
+    except WindowsError:
+        dpi_support = winreg.CreateKeyEx(
+            winreg.HKEY_CURRENT_USER,
+            r"Software\Microsoft\Internet Explorer\Main\FeatureControl\FEATURE_96DPI_PIXEL",
+            0, winreg.KEY_ALL_ACCESS)
+
+    mode = get_ie_mode()
+    executable_name = sys.executable.split("\\")[-1]
+    winreg.SetValueEx(browser_emulation, executable_name, 0, winreg.REG_DWORD,
+                      mode)
+    winreg.CloseKey(browser_emulation)
+
+    winreg.SetValueEx(dpi_support, executable_name, 0, winreg.REG_DWORD, 1)
+    winreg.CloseKey(dpi_support)
+    ie_updated = True
```

Comment:
I'm guessing the purpose of this is to ensure that the registry key is only set when the widget is first instantiated. However, it's still going to set the key once on every execution. 

Would it be better to:

1) Only set the key if it doesn't already exist (i.e., inside the try/catch block, only set the key if we needed to create it
2) Set this key as part of the installer, rather than at runtime?

Reply:
it sets key if user wants to use TogaWebBrowser
if we had set key already, it returns without working.(see line 22)

> 2. Set this key as part of the installer, rather than at runtime?

how can i do it?